src/init.cpp

```diff
@@ -155,7 +156,7 @@ static std::unique_ptr<ECCVerifyHandle> globalVerifyHandle;
 
 static std::thread g_load_block;
 
-static boost::thread_group threadGroup;
+static std::thread g_scheduler;
```

Comment:
It doesn't really make sense to run the scheduler without a thread, so I'd prefer to move this into the scheduler (like was done in #18710)

Reply:
Made some changes that should address this.